### PR TITLE
Fix for #2109: Generate ActiveQuery Class

### PR DESCRIPTION
**Pull request by klimov-paul**
Fix for https://github.com/yiisoft/yii2/issues/2109
Added ability to generate ActiveQuery class.

### CODE REVIEW

File: assets/gii.js [hunk 1]
```diff
@@ -146,24 +146,45 @@ yii.gii = (function ($) {
             initConfirmationCheckboxes();
             initToggleActions();
 
-            // model generator: hide class name input when table name input contains *
+            // model generator: hide class name inputs when table name input contains *
             $('#model-generator #generator-tablename').change(function () {
-                $('.field-generator-modelclass').toggle($(this).val().indexOf('*') == -1);
+                var show = ($(this).val().indexOf('*') == -1);
```
Comment (samdark) [inline assets/gii.js]: `==` → `===`

File: assets/gii.js [hunk 2]
```diff
@@ -146,24 +146,45 @@ yii.gii = (function ($) {
...
+                $('.field-generator-modelclass').toggle(show);
+                $('.field-generator-queryclass').toggle(show);
             }).change();
 
             // model generator: translate table name to model class
             $('#model-generator #generator-tablename').on('blur', function () {
                 var tableName = $(this).val();
-                if ($('#generator-modelclass').val()=='' && tableName && tableName.indexOf('*') === -1){
+                if ($('#generator-modelclass').val()=='' && tableName && tableName.indexOf('*') === -1) {
```
Comment (samdark) [inline assets/gii.js]: `==` → `===`

File: assets/gii.js [hunk 3]
```diff
@@ -146,24 +146,45 @@ yii.gii = (function ($) {
...
                     });
-                    $('#generator-modelclass').val(modelClass);
+                    $('#generator-modelclass').val(modelClass).blur();
                 }
             });
 
+            // model generator: translate model class to query class
+            $('#model-generator #generator-modelclass').on('blur', function () {
+                var modelClass = $(this).val();
+                if (modelClass != '') {
```
Comment (samdark) [inline assets/gii.js]: `==` → `===`

File: assets/gii.js [hunk 4]
```diff
@@ -146,24 +146,45 @@ yii.gii = (function ($) {
...
+                    var queryClass = $('#generator-queryclass').val();
+                    if (queryClass == '') {
```
Comment (samdark) [inline assets/gii.js]: `==` → `===`

File: assets/gii.js (hunk 2 above)
Comment (samdark) [inline assets/gii.js]: In case of empty string it could be just `if(X)`

File: generators/model/Generator.php [hunk 5]
```diff
@@ -146,7 +161,8 @@ public function autoCompleteData()
      */
     public function requiredTemplates()
     {
-        return ['model.php'];
+        // @todo make 'query.php' before 2.1 release
```
Comment (samdark) [inline generators/model/Generator.php]: ?

File: generators/model/default/query.php [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+/**
+ * This is the template for generating the ActiveQuery class.
+ */
+
+/* @var $this yii\web\View */
+/* @var $generator yii\gii\generators\model\Generator */
+/* @var $className string class name */
+/* @var $modelClassName string related model class name */
+
+$modelFullClassName =$modelClassName;
```
Comment (samdark) [inline generators/model/default/query.php]: Space missing.

File: assets/gii.js (hunk 2 above)
Comment (klimov-paul) [inline assets/gii.js]: This is actually the orignal code. I have not modified it - only added a space before opening bracket.

File: assets/gii.js (hunk 1 above)
Comment (klimov-paul) [inline assets/gii.js]: This is actually the orignal code. I suppose it can be kept as it is

File: generators/model/Generator.php (hunk 5 above)
Comment (klimov-paul) [inline generators/model/Generator.php]: If I will make 'query.php' to be required, any custom 'model' template, which has been created before, will produce an exception on any usage. Leaving only 'model.php' allows those old templates to be used unless no one use query generation ability for them.
This is for BC keep.

File: assets/gii.js (hunk 2 above)
Comment (samdark) [inline assets/gii.js]: Worth modifying, I think.

File: generators/model/Generator.php (hunk 5 above)
Comment (samdark) [inline generators/model/Generator.php]: Ah, then comment should say `require query.php before 2.1 release`

**Comment by klimov-paul** [comment]
Code style fixed.
Synchronization between 'model ns' and 'query ns' added.

File: generators/model/Generator.php (hunk 5 above)
Comment (klimov-paul) [inline generators/model/Generator.php]: yes, fixied

**Pull request merged**